src/utils.rs

```diff
@@ -0,0 +1,245 @@
+#[cfg(target_os = "linux")]
+use crate::netlink;
+use crate::{check_command_output_status, Peer, WireguardInterfaceError};
+use std::{collections::HashSet, process::Command};
+
+#[cfg(target_os = "linux")]
+const DEFAULT_FWMARK_TABLE: u32 = 51820;
+
+/// Helper function to add routing.  
+#[cfg(target_os = "linux")]
+pub(crate) fn add_peers_routing(
+    peers: &[Peer],
+    ifname: &str,
+) -> Result<(), WireguardInterfaceError> {
+    debug!("Adding peers routing for interface: {ifname}");
+
+    let mut unique_allowed_ips = HashSet::new();
+    let mut default_route = None;
+    for peer in peers {
+        for addr in &peer.allowed_ips {
+            if addr.ip.is_unspecified() {
+                // Handle default route
+                default_route = Some(addr);
+                break;
+            }
+            unique_allowed_ips.insert(addr);
+        }
+    }
+
+    // If there is default route skip adding other routings.
+    if let Some(default_route) = default_route {
+        debug!("Found default route: {default_route:?}");
```

Comment:
This is misleading. What we found is a specific allowed IPs value and because of that we are adding a default route.